engine/src/test/java/org/entando/entando/aps/system/init/DatabaseManagerTest.java

```diff
@@ -0,0 +1,75 @@
+package org.entando.entando.aps.system.init;
+
+import java.io.IOException;
+
+import javax.sql.DataSource;
+
+import com.agiletec.aps.system.exception.ApsSystemException;
+import org.entando.entando.aps.system.init.cache.IInitializerManagerCacheWrapper;
+import org.entando.entando.aps.system.init.model.Component;
+import org.entando.entando.aps.system.init.model.SystemInstallationReport;
+import org.entando.entando.aps.system.init.util.ComponentUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.ListableBeanFactory;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.mockito.Mockito.when;
+
+public class DatabaseManagerTest {
+
+    private Component component;
+
+    @Mock
+    private IInitializerManagerCacheWrapper cacheWrapper;
+
+    @Mock
+    private ListableBeanFactory beanFactory;
+
+    @InjectMocks
+    private DatabaseManager databaseManager = null;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void should_uninstall_component() throws ApsSystemException, IOException {
+        component = ComponentUtils.getEntandoComponent("test_component");
+
+        when(beanFactory.getBeanNamesForType(DataSource.class)).thenReturn(new String[]{"portDataSource", "servDataSource"});
+
+        SystemInstallationReport systemInstallationReport = createMockReport();
+        databaseManager.uninstallComponentResources(component, systemInstallationReport);
```

Comment:
If you want you can wrap the fake report using the Mockito `spy()` method. Then you can use the `verify()` method on the object for checking specific calls performed by the tested method, like this:

```java
SystemInstallationReport systemInstallationReport = spy(createMockReport());
databaseManager.uninstallComponentResources(component, systemInstallationReport);
verify(systemInstallationReport, times(1)).removeComponentReport("test_component");
```